### ISSUE TITLE
Ways of saving an ML model

### ISSUE THREAD

**Issue by Emma-223** [Emma-223/LQ1_pocket-coffea#10]
See https://scikit-learn.org/stable/model_persistence.html

At first I tried `pickle`, and I ran into the problem that the pocket-coffea singularity container has older versions of awkward and pandas than what I had used for training. 

I have since moved to `onnx` because that seems to be what is used in the ttHbb example pocket-coffea analysis. This is currently working from the technical standpoint of being able to use `onnxruntime` in a dask worker plugin, but I am not completely sure that the model is being applied correctly to the events. As in, the model that I load with `onnxruntime` takes in a single array of floats and it is not clear to me how to make sure that it correctly identifies which column in the array belongs to which variable.

I would sort of prefer to go back to `pickle` because in that case the actual python object is preserved and I could use `scikit-learn` to do the application of the BDT, which I understand better and have found better documentation for. I think the environment problem could be solved (I have another open issue about running p-c with a venv), so it would then just be a matter of making my custom executor load the model with `pickle` instead of `onnxruntime`.

**Comment by Emma-223** [comment]
After further investigation I have determined the following:
In both cases, (onnx and sklearn loaded with pickle), the loaded model requires the columns to be in the same order as they were in the training array. It will not have the ability to infer the correct order based on variable names. So if we train with [sT, pt1stEle, pt2ndEle, Mee] then the array we evaluate in the p-c config needs to have the columns in the same order. The difference is that sklearn will read the column names and throw an error message if they are out of order. Onnx will just return nonsense predictions.

Having sorted this out, I compared the two for a set of 20 DY events and see that they agree in their output. The left column is probability of being background, while the right column is probability of signal. I think the solution here is to also have the training script save the list of columns (easy to do in pandas) so that we can use that in the p-c code to build the input array in the correct order. This also means that there would be no advantage to switching back to pickle, as I have now resolved my only real issue with using onnx.

<img width="483" alt="Image" src="https://github.com/user-attachments/assets/ee4b989f-36b3-4d90-a791-fb0f24ce8017" />
